src/main/java/in/ashwanthkumar/gocd/github/GitHubPRBuildPlugin.java

```diff
@@ -219,17 +219,21 @@ GoPluginApiResponse handleGetLatestRevision(GoPluginApiRequest goPluginApiReques
             GitHelper git = gitFactory.create(gitConfig, gitFolderFactory.create(flyweightFolder));
             git.cloneOrFetch(provider.getRefSpec());
             Map<String, String> branchToRevisionMap = git.getBranchToRevisionMap(provider.getRefPattern());
-            Revision revision = git.getLatestRevision();
             git.submoduleUpdate();
 
+            // Select the first PR
+            Map.Entry<String,String> entry = branchToRevisionMap.entrySet().iterator().next();
+            String key = entry.getKey();
+            Revision revision = git.getDetailsForRevision(branchToRevisionMap.get(key));
+            String branch = "gocd-pr/" + entry.getKey();
+
             Map<String, Object> response = new HashMap<>();
-            String defaultBranch = (StringUtils.isEmpty(gitConfig.getBranch())) ? "master" : gitConfig.getBranch();
-            Map<String, Object> revisionMap = populateRevisionMap(gitConfig, defaultBranch, revision);
+            Map<String, Object> revisionMap = populateRevisionMap(gitConfig, branch, revision);
             response.put("revision", revisionMap);
             Map<String, String> scmDataMap = new HashMap<>();
             scmDataMap.put(BRANCH_TO_REVISION_MAP, JSONUtils.toJSON(branchToRevisionMap));
```

Comment:
Yes that's the main scenario and sounds good. I'm thinking of the probably uncommon scenario where there are **multiple** PRs open before the SCM material is created. In that case, when we return the full map here, the other PRs will not be built since in subsequent calls to `handleGetLatestRevisionSince` the plugin compares to the state and thinks it's already seen and reported those other PRs.

Reply:
Ah, ok, now I see. Indeed, what you describe can happen, and gocd will not correctly pick up the other PRs created before the SCM. I changed the PR to remove other branches from the `branchToRevisionMap`.